### PR TITLE
move imports to top of file in test_frames

### PR DESCRIPTION
**Pull request by nstarman**
### Description
Clean up some imports in the file `coordinates/tests/test_frames.py`
To address comment https://github.com/astropy/astropy/pull/10624#discussion_r480408131

### CODE REVIEW

**Comment by nstarman** [comment]
Something I was wondering, why doesn't astropy use a codestyler pre-commit?
I use `Black` in my work and was thinking of adding `isort`.
Black minimizes git diffs and ensures PEP8 and code appearance uniformity.

**Comment by pllim** [comment]
Sometimes, there are reasons to use local import. But for this particular file, I am not sure. Perhaps the import itself triggers some behavior? I'll let the subpackage maintainers review.

**Comment by pllim** [comment]
> why doesn't astropy use a codestyler pre-commit?

Because we have a lot of old code and having something like that would generate unrelated diff for a file and make it hard to review. It can be done, but not after a round of manual cleaning.

tl;dr -- Technical debt.

**Comment by nstarman** [comment]
> It can be done, but not after a round of manual cleaning.

Makes sense, but why not run the whole codebase through `Black` or something similar? It should be a completely non-breaking diff.

**Comment by pllim** [comment]
>  It should be a completely non-breaking diff

Hah... Perhaps you have not touched `io.fits` before. 😉 

But more seriously, we have 1k other issues to worry about if we have time, so making old code look nice, while desirable, is low priority for now.

**Comment by nstarman** [comment]
So it's not a problem if I use a more stringent code styler for my PRs? Is there one I should use instead of Black?

**Comment by pllim** [comment]
> So it's not a problem if I use a more stringent code styler for my PRs?

If you limit the style to your own relevant diff or new code, should not be a problem. It only becomes hard to review if you also apply it to other code unrelated to the feature or bug you are addressing. It's a balancing act, there are no set rules.

> Is there one I should use instead of Black?

In the CI, we use `flake8` (a very lenient one) and we also have a `pep8speaks` bot. They are both not as aggressive as Black. But there is also no rule that says you cannot use Black, as long as you keep the above recommendation in mind.

Hope this makes sense. We apologize for any inconvenience caused!

**Comment by adrn** [comment]
I don't see an obvious reason that these imports should stay inside their tests, so this is probably fine! But maybe @eteq or @mhvk remember the reasoning?

**Review by mhvk — APPROVED** [review]
Thanks for the cleanup, definitely looks all good to me! (I'm still not quite sold on the "blacken" kind of organization - it feels made to minimize git changes rather than to maximize clarity - though I must admit I can see the point for the imports, so perhaps it is just a thing one gets used to...)

**Comment by mhvk** [comment]
Failures both unrelated...

**Comment by nstarman** [comment]
> it feels made to minimize git changes rather than to maximize clarity

It definitely minimizes git changes. I've personally found that I can work to make code more clear — by grouping arguments when Black would separate them — but the extra effort compared to having Sublime run Black on every save... It took me a week to retrain my eye to prefer "blackened" code.

**Review by adrn — APPROVED** [review]

**Comment by bsipocz** [comment]
FYI the code style guide is very clear on the usage of black, it was discussed lengthy in the past pro an con between the developers. The black output is decided to be less legible for code review, and thus it is strongly discourage not to use it, which is in line with the previous recommendation of keeping the same code style as the files/modules originally have: https://docs.astropy.org/en/latest/development/codeguide.html#coding-style-conventions. Overall common grounds need to be find between individual editor/style/coding preferences and maintenance needs of a large library. 

``The use of automatic code formatters (e.g., Black) is strongly discouraged in contributions to Astropy.``

Arguably the sentence is hidden away in the guide, so maybe with the new pair of eyes you could suggest where the ideal place for it would be. Maybe in the PR draft? It already suggest the link and to run `tox -e codestyle`, another sentence might be useful still?

**Comment by nstarman** [comment]
Thanks for the info. I'll avoid Black in my PRs in the future.
